### DISCUSSION TITLE
Plot scaling

### DISCUSSION THREAD

**Discussion by PBfordev** [Blake-Madden/Wisteria-Dataviz#37] [Q&A] [answered]
Hi Blake,

your library looks very impressive, based on the demo. I am still trying to take a good look at it, but unfortunately, I am not very bright so I have been struggling with many basic things and just can't get the things going.

So, I have few questions:
1. Can `Wisteria::Canvas` (or just a single chart) be set so, that it sticks to a certain size, regardless of its parent size (as it is with other wxWidgets controls or Excel chart in a worksheet)?
2. When the plot size changes, I noticed that the font size used in the plot changes as well: Can it be done so the font size is the same (and device-independent), regardless of the plot size whether shown on screen (as it is with other wxWidgets controls or Excel chart in a worksheet), exported as image, or printed?

Thanks!

**Comment by Blake-Madden** [comment, marked as answer]
Hi PB!

So, rather than hard-coded sizes, the library takes more of a "scale and proportions" approach to sizing.

1. When a canvas's parent is resized, then the canvas (and its children graphs) will resize along with it. However, I think you could force a canvas to always be a certain size via the sizer you place it in. `Canvas` also derives from `wxScrolled`, so the base `SetMaxSize()` and `SetMinSize()` from `wxWindow` might do the trick (maybe depending on which type of container it is in).

Canvas does have a separate "SetCanvasMinSize()" API that forces it to never go smaller than a certain size (otherwise the scrollbars go into effect). If the aforementioned suggested don't work, would a new `SetCanvasMaxSize()` be of use?

Somewhat related, you can specify the proportions of things on a canvas. For example, you can have a histogram taking up a third of the canvas's width and a line plot next to it taking up two-thirds of the canvas's width.  So this proportion method was the approach I went with rather than hard-coded sizes.

2. No, the font always scales alongside the size of the plot. For example, you add a title to a graph with a 72pt font and also use 72pt font for the axis labels. When you resize the graph (or zoom into it), the fonts will scale up to the same proportion as how much the graph was resized.  I originally tried keeping the font the same size, but when I would zoom then the title would look tiny. As it is now, the font *technically* is still 72pt, at least relative scale of the graph.

Axis labels font sizes are a little bit different from other text. The font size of them is a suggestion, as I sometimes need to scale them down a little bit to make them fit.  That way they don't overlap each other (at least not too extremely).

**Reply by PBfordev** [reply]
Thanks for the reply.

In my rather limited experience (Microsoft Excel, ggplot2 in R, charts.js), scaling the font size with the chart size (or changing it when zooming its part) is rather unusual, but I understand your reasoning.

**Comment by Blake-Madden** [comment]
So when I use this library to print to PDF, it is the same as ggplot2 when you use it with knitr to save to PDF.  Same for when you save to PNG or other raster formats.  It's just in the demo where you can change the window size that the graph then rescales things to fit the new size.

I goal of mine has been to replace my current system of ggplot2/knitr reporting that I do because it doesn't give me the flexibility and unusual graphs that I need.  It's been going nicely so far.
